cms/functions.php

```diff
@@ -0,0 +1,260 @@
+    $index = $recId - 1;
+    $projectIdArray = $array[$index];
+    $projectId = '';
+    $projectId .= $projectIdArray['id'];
+    return $projectId;
+}
+
+/* Doc Block
+ * Returns article from array content provided by database.
+ *
+ * @param $array array associative array provided by database.
+ *
+ * @return string the content of the arrays within the arrays pulled from the database.
+ */
+function returnArticle(array $array):string {
+    $article = '';
+
+    foreach ($array as $value) {
+        $article .= $value['aboutMeArticle']."\n"."\n";
+    }
+    return $article;
+}
+
+/* Doc Block
+ * Returns single paragraph from array content provided by database.
+ *
+ * @param $array array associative array provided by database.
+ *
+ * @return string the content of the arrays within the arrays pulled from the database.
+ */
+function returnParagraph(array $array, int $id): string{
+    $index = $id - 1;
+    $paragraphArray = $array[$index];
+    $paragraph = '';
+    $paragraph .= $paragraphArray['aboutMeArticle'];
+    return $paragraph;
+}
+
+/* Doc Block
+ * Returns an associative array with content from portfolio table from database.
+ *
+ * @param $db object database link and credentials PDO object.
+ *
+ * @return array returns associative array with arrays containing content from portfolio table from database.
+ */
+function getProjectInfo(PDO $db):array {
+    $projectInfoQuery = $db->prepare("SELECT `id`, `projectTitle`, `projectTitleLink`,`projectImage` FROM `portfolio` ");
+    $projectInfoQuery->execute();
+    return $projectInfo = $projectInfoQuery->fetchAll();
+}
+
+/* Doc Block
+ * Updates the values of a row in portfolio table in database.
+ *
+ * @param $db object database link and credentials PDO object.
+ * @param $projectTitle string used with post from form and passed in to update the portfolio table in database.
+ * @param $projectTitleLink string used with post from form and passed in to update the portfolio table in database.
+ * @param $projectImage string object used with post from form and passed in to update the portfolio table in database.
+ * @param $projectId string used with post from form and passed in to update the portfolio table in database.
+ *
+ * @return array returns associative array with arrays containing content from portfolio table from database.
+ */
+function updateProject(PDO $db, string $projectTitle, string $projectTitleLink, string $projectImage, string $projectId) {
+
+    $query = $db->prepare("UPDATE `portfolio` SET `projectTitle` = :projectTitle, `projectImage` = :projectImage, `projectTitleLink` = :projectTitleLink WHERE `id` = :projectId");
+
+    $query->bindParam (':projectTitle',$projectTitle);
+    $query->bindParam (':projectTitleLink',$projectTitleLink);
+    $query->bindParam (':projectImage',$projectImage);
+    $query->bindParam (':projectId',$projectId);
+
+    $query->execute();
+
+    $updatedInfo = $projectId.' '. $projectTitle.' '.$projectTitleLink.' '.$projectImage;
```

Comment:
this is a false positive - if your query fails then you would return data that hasnt actually gone in the db

Reply:
Changed to return result of function that gets data from the database after the query is executed.